### PR TITLE
Do not generate profile.xml

### PR DESCRIPTION
**Pull request by pavel-mikula-sonarsource**
`SonarAnalyzer.RuleDescriptorGenerator` generates
* `rules.xml` - consumed by Java plugins as an embedded resource) 
* `profile.xml` - not used, can be removed.

This is part of bigger cleanup around RSPEC handling.

### CODE REVIEW

File: analyzers/src/SonarAnalyzer.RuleDescriptorGenerator/Program.cs [hunk 1]
```diff
@@ -36,53 +36,18 @@ public static class Program
...
-        }
-
-        private static void WriteXmlDescriptorFiles(string rulePath, string profilePath, string lang)
-        {
-            var language = AnalyzerLanguage.Parse(lang);
-
-            var genericRuleDetails = RuleDetailBuilder.GetAllRuleDetails(language).ToList();
-            var ruleDetails = genericRuleDetails.Select(RuleDetail.Convert).ToList();
-
-            Directory.CreateDirectory(lang);
```
Comment (pavel-mikula-sonarsource) [inline analyzers/src/SonarAnalyzer.RuleDescriptorGenerator/Program.cs:62]: This section is preserved.

File: analyzers/src/SonarAnalyzer.RuleDescriptorGenerator/Program.cs [hunk 2]
```diff
@@ -36,53 +36,18 @@ public static class Program
...
-                }));
-
-            SerializeObjectToFile(filePath, root);
-        }
-
-        private static void WriteRuleDescriptorFile(string filePath, IEnumerable<RuleDetail> ruleDetails)
-        {
-            var root = new RuleDescriptorRoot();
-            root.Rules.AddRange(ruleDetails);
-            SerializeObjectToFile(filePath, root);
```
Comment (pavel-mikula-sonarsource) [inline analyzers/src/SonarAnalyzer.RuleDescriptorGenerator/Program.cs:85]: This section is preserved and shortened to a single line. The rest is deleted.

File: analyzers/src/SonarAnalyzer.RuleDescriptorGenerator/Program.cs [hunk 3]
```diff
@@ -36,53 +36,18 @@ public static class Program
...
-            var root = new QualityProfileRoot(language);
-            root.Rules.AddRange(ruleDetails
-                .Where(descriptor => descriptor.IsActivatedByDefault)
-                .Select(descriptor => new QualityProfileRuleDescriptor(language)
-                {
-                    Key = descriptor.Key
-                }));
-
-            SerializeObjectToFile(filePath, root);
-        }
```
Comment (pavel-mikula-sonarsource) [inline analyzers/src/SonarAnalyzer.RuleDescriptorGenerator/Program.cs:79]: Purpose of this PR is to remove this method. And all scaffolding around it.

File: analyzers/src/SonarAnalyzer.RuleDescriptorGenerator/Program.cs [hunk 4]
```diff
@@ -36,53 +36,18 @@ public static class Program
...
+                Directory.CreateDirectory(language.ToString());
+                SerializeObjectToFile(Path.Combine(language.ToString(), "rules.xml"), root);
+            }
+            else
+            {
+                Console.WriteLine("[AnalyzerLanguage: 'cs' for C#, 'vbnet' for VB.Net]");
+                Console.WriteLine("All files will be created by the application");
             }
-
-            WriteXmlDescriptorFiles("rules.xml", "profile.xml", args[0]);
```
Comment (andrei-epure-sonarsource) [inline analyzers/src/SonarAnalyzer.RuleDescriptorGenerator/Program.cs:47]: archeology: leftover of https://github.com/SonarSource/sonar-dotnet/pull/1416

**Pull request merged**